src/components/table-container-generator/table-container-generator.js

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { TableContainer, Table, TableBody, Paper } from '@material-ui/core';
+import { useSelector } from 'react-redux';
+import TableContainerHead from '../table-container-head';
+import TableContainerRow from '../table-container-row';
+
+import useStyles from './table-container-generator.styles';
+import TablePaginator from '../table-container-paginator';
+import { config } from '../../configs';
+
+const TableContainerGenerator = ({ tableTitles, tableItems, pagination }) => {
+  const { SMALL_SIZE, DEFAULT_SIZE } = config.tableSizes;
+  const classes = useStyles();
+  const dense = useSelector((Table) => Table.dense);
+
+  return (
+    <div>
+      <TableContainer className={classes.container} component={Paper}>
+        <Table
+          className={classes.table}
+          stickyHeader
+          size={dense ? SMALL_SIZE : DEFAULT_SIZE}
+        >
+          <TableContainerHead titles={tableTitles} />
+          <TableBody id='table-body'>{tableItems}</TableBody>
+        </Table>
+      </TableContainer>
+      {pagination && <TablePaginator />}
+    </div>
+  );
+};
+
+TableContainerGenerator.propTypes = {
+  pagination: PropTypes.bool,
+  tableTitles: PropTypes.arrayOf(PropTypes.string.isRequired),
+  tableItems: PropTypes.arrayOf(PropTypes.shape(PropTypes.element.isRequired))
+};
+
+TableContainerGenerator.defaultProps = {
+  pagination: false,
+  tableTitles: [''],
```

Comment:
Can we have a default value for the `tableTitles` an empty array?

Reply:
Yes, we can! Redone.